### ISSUE TITLE
NodeListOf#item misses return type "null"

### ISSUE THREAD

**Issue by danielrentz** [microsoft/TypeScript-DOM-lib-generator#1261]
Interface `NodeList` is correctly typed
```typescript
interface NodeList {
    item(index: number): Node | null;
}
```
but interface `NodeListOf` is missing the `null` type:
```typescript
interface NodeListOf<TNode extends Node> extends NodeList {
    item(index: number): TNode;
}
```
therefore TSC does not warn about missing null checks:
```typescript
const node = document.childNodes.item(999999).firstChild;
```

**Comment by orta** [comment]
Agree 👍🏻 open to changes here
